subsys/bluetooth/host/settings.h

```diff
@@ -18,3 +18,25 @@ int bt_settings_decode_key(const char *key, bt_addr_le_t *addr);
 void bt_settings_save_id(void);
 
 int bt_settings_init(void);
+
+
+/**
+ * @defgroup bt_settings_paths Bluetooth Host Settings paths
+ * @brief Settings API paths used by Bluetooth Host
+
+ * @{
+ */
+
+/** @brief Settings API path to Bluetooth Host subtree.
+ *
+ *  @warning This value may be hardcoded and part of ABIs. Changing it will
+ *  likely break other code.
+ */
+#define BT_SETTINGS_SUBTREE "bt"
```

Comment:
I’m not really convinced of the usefulness or value of this change. “bt” is as much of a namespace for Bluetooth settings as BT_* and bt_* are for Bluetooth APIs. These need to stay consistent with each other and I think it’s unlikely we’ll ever change them. Additionally, the new macro is much longer to write than “bt” 🙂

Reply:
It was me that requested it, so I'll elaborate on why:

In my opinion when you have a value (e.g. `"bt"`) that is used and compared multiple places, it should never be a literal value. Not only do you lose all semantic meaning when using a literal, but you also introduce risk of missing one (of possible many) occurrences when refactoring (even if it is considered a static value). 

Even though the macro is longer, my opinion is still that it increases readability, because you know that any use of "bt" (when using the macro) is relevant to the SETTINGS.

Reply:
The current code is a bit unkempt in my opinion, so here's a suggestion we came up with @alwa-nordic and @jhedberg:

```
#define BT_SETTINGS(subpath) "bt/" subpath
```

and then, it could be used with:

```
SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh, BT_SETTINGS("mesh"), NULL, NULL, mesh_commit,
			       NULL);
```

Reply:
Regardless, this should not be part of this PR, but a separate one. Can we merge this first, removing the unrelated changes first, and then have anyone who is interested present an RFC PR where we can discuss this in detail.